contracts/cw20-atomic-swap/src/contract.rs

```diff
@@ -622,8 +684,121 @@ mod tests {
                 recipient: create2.recipient,
                 source: sender2,
                 expires: create2.expires,
-                balance
+                balance: BalanceHuman::Native(balance),
             }
         );
     }
+
+    #[test]
```

Comment:
In general, looks good. I need to read through this again to make sure all the test cases are covered.

Reply:
Yes... one of the advantages of joining both coins in one set of functions is that most of the tests for invalid values / checks work for both coins, and so have to be written only once.

I'll recheck everything too, just to err on the safe side.

Reply:
> most of the tests for invalid values / checks work for both coins, and so have to be written only once.

This is key and helps avoid not just duplicate code but bugs. In fact the opyn hack was based on them testing the erc20 path extensively but not the ETH path. I think making helpers that unify cw20 and native coins is super useful to both make it easier to build and to avoid such bugs